diffrax/_solver/foster_langevin_srk.py

```diff
@@ -0,0 +1,410 @@
+import abc
+from collections.abc import Callable
+from typing import Any, Generic, TypeVar
+
+import equinox as eqx
+import equinox.internal as eqxi
+import jax
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from equinox import AbstractVar
+from jax import vmap
+from jaxtyping import PyTree
+
+from .._custom_types import (
+    AbstractBrownianIncrement,
+    BoolScalarLike,
+    DenseInfo,
+    RealScalarLike,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._solution import RESULTS
+from .._term import (
+    AbstractTerm,
+    MultiTerm,
+    UnderdampedLangevinDiffusionTerm,
+    UnderdampedLangevinDriftTerm,
+    UnderdampedLangevinLeaf,
+    UnderdampedLangevinTuple,
+    UnderdampedLangevinX,
+    WrapTerm,
+)
+from .base import AbstractItoSolver, AbstractStratonovichSolver
+
+
+_ErrorEstimate = TypeVar("_ErrorEstimate", None, UnderdampedLangevinTuple)
+ULDArgs = tuple[
```

Comment:
Should this be `UDL` or `ULD`?

That aside, personally I prefer to use the more-verbose approach to things, i.e. `UnderdampedLangevinArgs`. I assume everyone is using an editor with autocomplete these days.

Reply:
It's ULD for Underdamped Langevin Diffusion. But yes, I can go with the longer version.